client/components/ProductOverview/layouts/ImageGallery.js

```diff
@@ -1,79 +1,101 @@
-      setCaroulImgs([
-        ...caroulImgs,
-        <img height="630px" width="800px" src={photo.url} />,
-      ]);
-    });
-  };
-
-  useEffect(() => {
-    getCaroulImgs();
-  }, [curStyle]);
-
-  const [slides, setSlides] = useState([
-    <img
-      height="630px"
-      width="800px"
-      src={
-        'https://i.picsum.photos/id/482/200/200.jpg?hmac=_oPafYMoTEVlnjtzFvYD5JoiK6Q-xfxbki4tYEMuGRI'
-      }
-    />,
-    <img
-      height="630px"
-      width="800px"
-      src={
-        'https://i.picsum.photos/id/577/200/200.jpg?hmac=o426LHIb2MgD4TJnSfBnhpGdOTOdofsp1A7yraIpo-Y'
-      }
-    />,
-    <img
-      height="630px"
-      width="800px"
-      src={
-        'https://i.picsum.photos/id/61/200/200.jpg?hmac=RL-JLeHTLsWK7354qjYNQ1iuDxeoCv-kYRTUYun2h34'
-      }
-    />,
-    <img
-      height="630px"
-      width="800px"
-      src={
-        'https://i.picsum.photos/id/58/200/200.jpg?hmac=aol3E3KC2fpsVXlPhgxLR9-CLoUQa-kbswhZx-gYzCE'
-      }
-    />,
-    <img
-      height="630px"
-      width="800px"
-      src={
-        'https://i.picsum.photos/id/622/200/200.jpg?hmac=0opC4wvaKSUqImE8atOt5HC8k6S4bXipDuItdfzK9s4'
-      }
-    />,
-  ]);
-
-  const updateVal = (v) => [setCaroulVal(v)];
+  const [curSlide, setCurSlide] = useState(0);
 
   return (
-    <div className="mGalleryBody">
-      {/* <Carousel value={caroulVal} slides={slides} onChange={setCaroulVal} /> */}
-      {/* <div className="mSubImages">
-        <div></div>
-        <div></div>
-        <div></div>
-        <div></div>
-        <div></div>
-        <div></div>
-        <div></div>
-        <div></div>
-      </div> */}
-    </div>
+    <CarouselProvider
+      totalSlides={curStyle.photos.length}
+      naturalSlideHeight={100}
+      naturalSlideWidth={100}
+      dragEnabled={false}
+    >
+      <div style={{ height: '700px', width: '700px', position: 'relative' }}>
+        <div
+          className="position-absolute row h-100 w-100"
+          style={{ zIndex: '1' }}
+        >
+          <div className="col-2">
+            <div className="col p-3">
+              {curStyle.photos.map((thumbnail, i) => (
+                <Dot
+                  onClick={() => setCurSlide(i)}
+                  className="mCarouselStyle"
+                  slide={i}
+                  key={i}
+                >
+                  <div
+                    className="bg-secondary my-3"
+                    style={{
+                      height: '60px',
+                      width: '60px',
+                      backgroundImage: `url(${thumbnail.thumbnail_url})`,
+                      opacity: curSlide === i ? '' : '50%',
+                      border: curSlide === i && '2px solid black',
+                    }}
+                  ></div>
+                </Dot>
+              ))}
+            </div>
+          </div>
+          <div className="col-10 d-flex justify-content-between align-items-center">
+            {curSlide !== 0 ? (
```

Comment:
Instead of using ternary operator with a null div, I think using this pattern might work better: (boolean expression && conditionally rendered JSX)

Makes more sense semantically

Reply:
I use ternary instead of boolean expression because I need something to be there. Its part of my grid/flex, if not then the placement will be wrong.

Reply:
Gotcha, that's fine then.